src/constructs/aws/abstracts/StaticWebsiteAbstract.ts

```diff
@@ -0,0 +1,316 @@
+        }
+
+        getUtils().log(
+            `Emptying S3 bucket '${bucketName}' for the '${this.id}' static website, else CloudFormation will fail (it cannot delete a non-empty bucket)`
+        );
+        await emptyBucket(this.provider, bucketName);
+    }
+
+    async getUrl(): Promise<string | undefined> {
+        const domain = await this.getDomain();
+        if (domain === undefined) {
+            return undefined;
+        }
+
+        return `https://${domain}`;
+    }
+
+    async getBucketName(): Promise<string | undefined> {
+        return this.provider.getStackOutput(this.bucketNameOutput);
+    }
+
+    async getDomain(): Promise<string | undefined> {
+        return this.provider.getStackOutput(this.domainOutput);
+    }
+
+    async getCName(): Promise<string | undefined> {
+        return this.provider.getStackOutput(this.cnameOutput);
+    }
+
+    async getDistributionId(): Promise<string | undefined> {
+        return this.provider.getStackOutput(this.distributionIdOutput);
+    }
+
+    private errorResponseDocument(): string {
+        // Custom error page
+        if (this.configuration.errorPage !== undefined) {
+            const errorPath = this.configuration.errorPage;
+            if (errorPath.startsWith("./") || errorPath.startsWith("../")) {
+                throw new ServerlessError(
+                    `The 'errorPage' option of the '${this.id}' static website cannot start with './' or '../'. ` +
+                        `(it cannot be a relative path).`,
+                    "LIFT_INVALID_CONSTRUCT_CONFIGURATION"
+                );
+            }
+
+            return errorPath;
+        }
+
+        /**
+         * The default behavior is optimized for SPA: all unknown URLs are served
+         * by index.html so that routing can be done client-side.
+         */
+        return "index.html";
+    }
+
+    private createViewerResponseCloudfrontFunction(): cloudfront.Function {
+        const securityHeaders: Record<string, { value: string }> = {
+            "x-frame-options": { value: "SAMEORIGIN" },
+            "x-content-type-options": { value: "nosniff" },
+            "x-xss-protection": { value: "1; mode=block" },
+            "strict-transport-security": { value: "max-age=63072000" },
+        };
+        if (this.configuration.security?.allowIframe === true) {
+            delete securityHeaders["x-frame-options"];
+        }
+        const jsonHeaders = JSON.stringify(securityHeaders, undefined, 4);
+        /**
+         * CloudFront function that manipulates the HTTP responses to add security headers.
+         */
+        const code = `function handler(event) {
+    var response = event.response;
+    response.headers = Object.assign({}, ${jsonHeaders}, response.headers);
+    return response;
+}`;
+
+        return new cloudfront.Function(this, "ViewerResponseFunction", {
+            functionName: `${this.provider.stackName}-${this.provider.region}-${this.id}-viewerResponse`,
```

Comment:
Small nitpick but can we keep this one named `ResponseFunction`? That will minimize diffs/changes in tests, and in CloudFront Functions there is not "Origin" response function so the distinction is not needed. So anything that lets us have simpler naming/concepts is welcome.

```suggestion
        return new cloudfront.Function(this, "ResponseFunction", {
            functionName: `${this.provider.stackName}-${this.provider.region}-${this.id}-response`,
```

Reply:
Sure no problem !

The reason I made this refactoring in the first place is because I started implemeting a solution with a Lamda@Edge triggered by the origin response trigger, thus I needed the function names to be really explicit.